include/beman/exemplar/identity.hpp

```diff
@@ -27,16 +27,14 @@ namespace beman::exemplar {
 struct __is_transparent; // not defined
 
 // A function object that returns its argument unchanged.
-struct identity
-{
+struct identity {
     // Returns `t`.
     template <class T>
 #if defined(__cpp_constexpr)
```

Comment:
I realize this is just a formatting PR, but I'm wondering if we actually need this macro for Beman? This would imply support for something prior to c++11 which really isn't our aim here.

Reply:
Yes this is a formatting pr! I am just presenting a preview of the formatting that is going to be adopted.

This is a good point! There's also a macro test for `__cpp_lib_transparent_operators` at testing implying C++14.

Reply:
yes, well I guess I could see some library want c++14 support, but 98 -- not really.  I suggest we move the macro at 33 as a drive by fix on this PR.

Reply:
That's fair! After the weekly sync yesterday I think I will expand the scope of this pr out to be "Introduce general pre-commit + linting", haven't got around to updating the title and description, so I will need to work on this for a bit, I will open a separate pr specifically deleting this line.

Reply:
Ok I'm good with that

Reply:
As another note you'll see in optional26 PR the unadorned use of constexpr  https://github.com/beman-project/optional26/pull/64/files

Reply:
`unadorned`, cool new vocab for me!

I think I will actually hold creating a pr regarding this in hope of `clang-tidy` flagging it.

Reply:
May be bundled into #38

Reply:
Fix in #55